### PR TITLE
Add rstack

### PR DESCRIPTION
**Pull request by DiamondTaco**
i use a lot of switch cases in the implementation, so im open to another solution if there is one.

### CODE REVIEW

File: src/main/java/com/domain/redstonetools/features/arguments/DirectionArgumentType.java [hunk 1]
```diff
@@ -41,12 +41,11 @@ public static Direction firstOrdinal(Direction playerFacing) {
...
     // big evil match direction function, there might be a better way to do this but i don't know how
     @Nullable
-    public static Direction matchDirection(String direction, Direction playerFacing) throws CommandSyntaxException {
+    public static Direction matchDirection(String direction, Direction playerFacing) throws CommandException {
         return switch (direction) {
             case "me", "forward" -> firstOrdinal(playerFacing);
             case "left" -> switch (firstOrdinal(playerFacing)) {
-                case UP, DOWN ->
-                        throw new CommandSyntaxException(null, new LiteralMessage("Can't determine direction"));
+                case UP, DOWN -> throw new CommandException(Text.of("Can't determine direction"));
```
[the remark this answers is not in the document]
Comment (DiamondTaco) [reply src/main/java/com/domain/redstonetools/features/arguments/DirectionArgumentType.java:48]: if you are facing straight up or down i can't figure out what direction to spawn the stack in, but for back i can just flip direction. returning the same direction was a bug tho

[the remark this answers is not in the document]
Comment (DiamondTaco) [reply src/main/java/com/domain/redstonetools/features/arguments/DirectionArgumentType.java:48]: fixed

**Review by Matthias1590 — CHANGES_REQUESTED** [review]

File: src/main/java/com/domain/redstonetools/features/arguments/DirectionArgumentType.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import com.sk89q.worldedit.math.BlockVector3;
+import com.sk89q.worldedit.util.Direction;
+import net.minecraft.command.CommandException;
+import net.minecraft.text.Text;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Set;
+
+public class DirectionArgumentType extends SetArgumentType<String> {
+    private static final Set<String> DIRS = Set.of(
```
[the remark this answers is not in the document]
Comment (DiamondTaco) [reply src/main/java/com/domain/redstonetools/features/arguments/DirectionArgumentType.java:12]: We can't use a we class bc they use a parser to parse directions, not just a set of strings. I think its better to leave this as a set of strings, this is just used to accept directional input and WE already has a direction enum for dealing with actual directions

[the remark this answers is not in the document]
Comment (Matthias1590) [reply src/main/java/com/domain/redstonetools/features/arguments/DirectionArgumentType.java:12]: > We can't use a we class bc they use a parser to parse directions, not just a set of strings. I think its better to leave this as a set of strings, this is just used to accept directional input and WE already has a direction enum for dealing with actual directions

An enum is less error prone and would remove the need for half of the code in this feature

[the remark this answers is not in the document]
Comment (DiamondTaco) [reply src/main/java/com/domain/redstonetools/features/arguments/DirectionArgumentType.java:12]: > > We can't use a we class bc they use a parser to parse directions, not just a set of strings. I think its better to leave this as a set of strings, this is just used to accept directional input and WE already has a direction enum for dealing with actual directions
> 
> An enum is less error prone and would remove the need for half of the code in this feature

not really, most of the matching is parsing the directions directly. unless you want to remove features like "//rstack 5 up" you have to live with with the boilerplate (AFAIK)

[the remark this answers is not in the document]
Comment (Matthias1590) [reply src/main/java/com/domain/redstonetools/features/arguments/DirectionArgumentType.java:12]: Enums can get rid of most boilerplate in this case

[the remark this answers is not in the document]
Comment (DiamondTaco) [reply src/main/java/com/domain/redstonetools/features/arguments/DirectionArgumentType.java:12]: well i don't know how they would, can you provide an example on how they would match relative directions like left, back, etc?

File: src/main/java/com/domain/redstonetools/features/arguments/DirectionSerializer.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import com.domain.redstonetools.utils.DirectionArgument;
+
+public class DirectionSerializer extends EnumSerializer<DirectionArgument> {
+    private static final DirectionSerializer INSTANCE = new DirectionSerializer();
+
+    private DirectionSerializer() {
+        super(DirectionArgument.class);
+    }
+
+    public static DirectionSerializer directionArgument() {
```
Comment (Matthias1590) [inline src/main/java/com/domain/redstonetools/features/arguments/DirectionSerializer.java:12]: Could be renamed to direction

**Review by Matthias1590 — CHANGES_REQUESTED** [review]

File: src/main/java/com/domain/redstonetools/features/arguments/DirectionSerializer.java (hunk 3 above)
Comment (DiamondTaco) [reply src/main/java/com/domain/redstonetools/features/arguments/DirectionSerializer.java:12]: which one? DirectionSerializer or directionArgument()

Comment (Matthias1590) [reply src/main/java/com/domain/redstonetools/features/arguments/DirectionSerializer.java:12]: directionArgument

Comment (DiamondTaco) [reply src/main/java/com/domain/redstonetools/features/arguments/DirectionSerializer.java]: done

**Pull request merged**